### PR TITLE
feat: upgrade ms-authentication service to edgex 2.2

### PR DESCRIPTION
**Pull request by vli11**
Signed-off-by: Valina Li <valina.li@intel.com>

### CODE REVIEW

File: ms-authentication/routes/common.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Copyright © 2020 Intel Corporation. All rights reserved.
+// Copyright © 2022 Intel Corporation. All rights reserved.
```
Comment (jim-wang-intel) [inline ms-authentication/routes/common.go:1]: same here

File: ms-authentication/routes/gets.go [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// Copyright © 2020 Intel Corporation. All rights reserved.
+// Copyright © 2022 Intel Corporation. All rights reserved.
```
Comment (jim-wang-intel) [inline ms-authentication/routes/gets.go:1]: same here

File: ms-authentication/routes/models.go [hunk 3]
```diff
@@ -1,4 +1,4 @@
-// Copyright © 2020 Intel Corporation. All rights reserved.
+// Copyright © 2022 Intel Corporation. All rights reserved.
```
Comment (jim-wang-intel) [inline ms-authentication/routes/models.go:1]: same here

**Comment by lenny-intel** [comment]
@vli11 please update the commit message and PR title with better descriptive. 

i.e "upgrade authentication" doesn't match these changes.

**Comment by vli11** [comment]
> @vli11 please update the commit message and PR title with better descriptive.
> 
> i.e "upgrade authentication" doesn't match these changes.

updated!

File: ms-authentication/Dockerfile [hunk 4]
```diff
@@ -1,10 +1,10 @@
-# Copyright © 2020 Intel Corporation. All rights reserved.
+# Copyright © 2022 Intel Corporation. All rights reserved.
```
[the remark this answers is not in the document]
Comment (brian-intel) [reply ms-authentication/Dockerfile]: I believe you're correct Jim. If we didn't make any code changes then we aren't required to update the license year.

**Review by lenny-intel — CHANGES_REQUESTED** [review]

File: ms-authentication/main.go [hunk 5]
```diff
@@ -1,49 +1,47 @@
...
-	// Create an instance of the EdgeX SDK and initialize it.
-	edgexSdk := &appsdk.AppFunctionsSDK{ServiceKey: serviceKey}
-	if err := edgexSdk.Initialize(); err != nil {
-		edgexSdk.LoggingClient.Error(fmt.Sprintf("SDK initialization failed: %v\n", err))
-		os.Exit(-1)
-	}
+type AuthApp struct {
+	service interfaces.ApplicationService
+	lc      logger.LoggingClient
+}
```
Comment (lenny-intel) [inline ms-authentication/main.go:23]: Not needed. Left over from wrapper

File: ms-authentication/res/configuration.toml [hunk 6]
```diff
@@ -1,43 +1,27 @@
...
-
-[Binding]
+[Trigger]
 Type="http"
-SubscribeTopic=""
-PublishTopic=""
 
 [ApplicationSettings]
+
+[AppCustom]
```
Comment (lenny-intel) [inline ms-authentication/res/configuration.toml:27]: Remove. Left over...

**Review by lenny-intel — CHANGES_REQUESTED** [review]

**Review by jim-wang-intel — COMMENTED** [review]
LGTM upon Lenny's approval

**Review by brian-intel — APPROVED** [review]
LGTM pending changes

File: ms-authentication/main.go (hunk 5 above)
Comment (vli11) [reply ms-authentication/main.go:23]: we still need AuthApp for main to create app

File: ms-authentication/Dockerfile [hunk 7]
```diff
@@ -26,4 +26,4 @@ COPY --from=builder /usr/local/bin/ms-authentication/cards.json /cards.json
 COPY --from=builder /usr/local/bin/ms-authentication/accounts.json /accounts.json
 COPY --from=builder /usr/local/bin/ms-authentication/people.json /people.json
 
-CMD [ "/ms-authentication", "--confdir=/res", "-cp=consul.http://edgex-core-consul:8500", "-r"]
+CMD [ "/ms-authentication", "--confdir=/res", "-cp=consul.http://edgex-core-consul:8500", "-r -s"]
```
Comment (jim-wang-intel) [inline ms-authentication/Dockerfile:29]: standard way is to separate "-r", "-s" into different arguments

File: docker-build-image/go.mod [hunk 8]
```diff
@@ -3,25 +3,25 @@
```
Comment (jim-wang-intel) [inline docker-build-image/go.mod:3]: need to change copyright year for this one

File: docker-compose.edgex.yml [hunk 9]
```diff
@@ -0,0 +1,197 @@
+# Copyright © 2020-2022 Intel Corporation. All rights reserved.
```
Comment (lenny-intel) [inline docker-compose.edgex.yml:1]: These compose file updates should be in a separate PR.

File: docker-compose.ac.yml [hunk 10]
```diff
@@ -0,0 +1,40 @@
+# Copyright © 2020-2022 Intel Corporation. All rights reserved.
```
Comment (lenny-intel) [inline docker-compose.ac.yml:1]: I am good with this compose file snippet in this PR, but not the full EdgeX one which goes beyond scope of this PR.

File: docker-compose.ac.yml [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+      command:
+        condition: service_started
+    environment:
+      # Comment out and/or update the setting of http_proxy if needed (e.g. not using proxy)
+      CLIENTS_CORE_DATA_HOST: edgex-core-data
+      CLIENTS_CORE_METADATA_HOST: edgex-core-metadata
+      CLIENTS_SUPPORT_SCHEDULER_HOST: edgex-support-scheduler
+      DATABASES_PRIMARY_HOST: edgex-redis
+      EDGEX_SECURITY_SECRET_STORE: "false"
+      # MESSAGEQUEUE_HOST: edgex-redis
```
Comment (lenny-intel) [inline docker-compose.ac.yml:22]:
```suggestion
      EDGEX_SECURITY_SECRET_STORE: "false"
```

File: ms-authentication/main.go (hunk 5 above)
Comment (lenny-intel) [reply ms-authentication/main.go:23]: main shouldn't be creating app. Doesn't add anything but more complexity. Keep it simple :-)

File: ms-authentication/main.go [hunk 12]
```diff
@@ -1,49 +1,47 @@
...
-	appSettings := edgexSdk.ApplicationSettings()
-	if appSettings == nil {
-		edgexSdk.LoggingClient.Error("No application settings found")
-		os.Exit(-1)
+func main() {
+	// TODO: See https://docs.edgexfoundry.org/2.2/microservices/application/ApplicationServices/
+	//       for documentation on application services.
+	app := AuthApp{}
+	var ok bool
+	app.service, ok = pkg.NewAppService(serviceKey)
```
Comment (lenny-intel) [inline ms-authentication/main.go:30]:
```suggestion
	service, ok := pkg.NewAppService(serviceKey)
```

File: ms-authentication/main.go [hunk 13]
```diff
@@ -1,49 +1,47 @@
...
+	if !ok {
+		os.Exit(1)
 	}
+	app.lc = app.service.LoggingClient()
```
Comment (lenny-intel) [inline ms-authentication/main.go:34]:
```suggestion
	lc := app.service.LoggingClient()
```

File: ms-authentication/main.go [hunk 14]
```diff
@@ -1,49 +1,47 @@
...
 
-	if err := edgexSdk.AddRoute("/authentication/{cardid}", routes.AuthenticationGet, "GET"); err != nil {
-		edgexSdk.LoggingClient.Error(fmt.Sprintf("Unable to add /authentication/{cardid} GET route: %v\n", err))
-		os.Exit(-1)
+	if err := app.service.AddRoute("/authentication/{cardid}", routes.AuthenticationGet, "GET"); err != nil {
```
Comment (lenny-intel) [inline ms-authentication/main.go:36]:
```suggestion
	if err := service.AddRoute("/authentication/{cardid}", routes.AuthenticationGet, "GET"); err != nil {
```

File: ms-authentication/main.go [hunk 15]
```diff
@@ -1,49 +1,47 @@
...
+		app.lc.Errorf("Unable to add /authentication/{cardid} GET route: %s", err.Error())
```
Comment (lenny-intel) [inline ms-authentication/main.go:37]:
```suggestion
		lc.Errorf("Unable to add /authentication/{cardid} GET route: %s", err.Error())
```

File: ms-authentication/main.go [hunk 16]
```diff
@@ -1,49 +1,47 @@
...
+		os.Exit(1)
 	}
 
-	// Tell the SDK to "start" and begin listening for events to trigger the pipeline.
-	err := edgexSdk.MakeItRun()
-	if err != nil {
-		edgexSdk.LoggingClient.Error("MakeItRun returned error: ", err.Error())
-		os.Exit(-1)
+	if err := app.service.MakeItRun(); err != nil {
```
Comment (lenny-intel) [inline ms-authentication/main.go:41]:
```suggestion
	if err := service.MakeItRun(); err != nil {
```

File: ms-authentication/main.go [hunk 17]
```diff
@@ -1,49 +1,47 @@
...
+		app.lc.Errorf("MakeItRun returned error: %s", err.Error())
```
Comment (lenny-intel) [inline ms-authentication/main.go:42]:
```suggestion
		lc.Errorf("MakeItRun returned error: %s", err.Error())
```

File: docker-compose.edgex.yml (hunk 9 above)
Comment (vli11) [reply docker-compose.edgex.yml:1]: > These compose file updates should be in a separate PR.

it was in separated PR and already merged into Edgex2.2 branch, this happens to be here because I did a rebase of Edgex2.2

**Review by lenny-intel — CHANGES_REQUESTED** [review]

File: docker-compose.edgex.yml (hunk 9 above)
Comment (lenny-intel) [reply docker-compose.edgex.yml:1]: A rebase should exclude it.

File: docker-compose.ac.yml [hunk 18]
```diff
@@ -0,0 +1,40 @@
...
+        condition: service_started
+      command:
+        condition: service_started
+    environment:
+      # Comment out and/or update the setting of http_proxy if needed (e.g. not using proxy)
+      CLIENTS_CORE_DATA_HOST: edgex-core-data
+      CLIENTS_CORE_METADATA_HOST: edgex-core-metadata
+      CLIENTS_SUPPORT_SCHEDULER_HOST: edgex-support-scheduler
+      DATABASES_PRIMARY_HOST: edgex-redis
+      # MESSAGEQUEUE_HOST: edgex-redis
```
Comment (lenny-intel) [inline docker-compose.ac.yml:21]: Remove these unused settings and unneeded comments

**Review by lenny-intel — CHANGES_REQUESTED** [review]

**Review by brian-intel — APPROVED** [review]

**Review by lenny-intel — APPROVED** [review]
LGTM
